### ISSUE TITLE
Can't specify `data` parameter explicitly in instantiation of NDDataArray

### ISSUE THREAD

**Issue by evertrol** [astropy/astropy#4620]
`NDDataArray` doesn't allow me to pass its first (and only) positional argument, `data`, explicitly, like
```python
>>> NDDataArray(data=np.ones((2,2)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/site-packages/astropy/nddata/compat.py", line 101, in __init__
    data = arg[0]  # do this after parent __init__, to validate the args
IndexError: tuple index out of range
```
The reason is the way [`NDDataArray`s `__init__`](https://github.com/astropy/astropy/blob/master/astropy/nddata/compat.py#L84) is set up, in particular the line from the traceback above:
`data = arg[0]`
doesn't work, since arg(*) is empty, and `data` is already its own variable.

I think it's nicer if one can occasionally be explicit about the arguments naming, and it allows one to mix positional and keyword arguments (for ease of use, if one has forgotten the order). `NDData` allows this:
```python
>>> NDData(np.ones((2,2)), meta={})
NDData([[ 1.,  1.],
        [ 1.,  1.]])
>>> NDData(meta={}, data=np.ones((2,2)))
NDData([[ 1.,  1.],
        [ 1.,  1.]])
```

I'll put in a PR later, unless this is vetoed before that.

(\*) it is one argument, but shouldn't that really be just `*args`?

**Comment by MSeifert04** [comment]
I think having ``data`` as named parameter would be nice. If you haven't changed it yet, maybe you could try changing it to ``data = arg[0] if arg else kwd['data']`` (haven't tested it). Also you maybe need to test it against some affiliated packages because ``NDDataArray`` is used as superclass for example in ``ccdproc``.

I agree with ``args`` instead of ``arg``. But doesn't make any difference either way and in my opinion "never change a winning team" but some may disagree.

**Comment by mwcraig** [comment]
I'll dig into the history later this week..I remember merging the PR that made this change, don't remember what the underlying reasoning was. The PR was #4130

**Comment by evertrol** [comment]
I've put in a PR. It is a bit more explicit with the `ndata` keyword in `__init__`. @MSeifert04 's suggestion keeps the usual `*args, **kwargs` structure intact, so that might be preferred.
This issue actually came up using ccdproc, so I've tested it against ccdproc as well, and that seems to be ok. Could use an extra check, since that won't be tested using CI.

**Comment by mwcraig** [comment]
Closed by #4626
